content/introduction/how-ipfs-works.md

```diff
@@ -0,0 +1,75 @@
+
+Content addressing through hashes has become a widely-used means of connecting data in distributed systems, from the commits that back your code to the blockchains that run cryptocurrencies. However, the underlying data structures in these systems are not necessarily interoperable.
+
+This is where the [IPLD project](https://ipld.io/) comes in. **Hashes identify content, and IPLD translates between data structures**. Since different distributed systems structure their data in different ways, IPLD provides libraries for combining pluggable modules (parsers for each possible type of IPLD node) to resolve a path, selector, or query across many linked nodes (allowing you explore data regardless of the underlying protocol). IPLD provides a way to translate between content-addressable data structures: “Oh you use Git-style, no worries, I can follow those links. Oh you use Ethereum, I got you, I can follow those links too!”
+
+The IPFS protocol uses “IPFS-flavored IPLD” to get from raw content to an IPFS address. IPFS has its own preferences and conventions about how data should be broken up into a DAG (more on DAGs below!); IPLD links content on the IPFS network together using those conventions.
+
+**Everything else in the IPFS ecosystem builds on top of this core concept: linked, addressable content is the fundamental connecting element that makes the rest work.**
+
+### 2 \\ DAGs
+
+IPFS and many other distributed systems take advantage of a data structure called [directed acyclic graphs](https://en.wikipedia.org/wiki/Directed_acyclic_graph), or DAGs. Specifically, they use _Merkle-DAGs_, which are DAGs where each node has an identifier that is a hash of the node’s contents. Sound familiar? This refers back to the _CID_ concept that we covered in the previous section. Another way to look the this CID-linked-data concept: identifying a data object (like a Merkle-DAG node) by the value of its hash is _content addressing_. _(Check out [the concept guide on Merkle-DAGs]({{<relref "guides/concepts/merkle-DAG.md">}}) for a more in-depth treatment of this topic.)_
+
+IPFS uses a Merkle-DAG that is optimized for representing directories and files, but you can structure a Merkle-DAG in many different ways. For example, Git uses a Merkle-DAG that has many versions of your repo inside of it.
+
+To build a Markle-DAG representation of your content, IPFS often first splits it into _blocks_. If you had a huge file with a single CID, you’d have to transfer the whole file in a single chunk every time someone requested it. But instead, you can break it into blocks, and transfer it block by block. Splitting it into blocks also means that different parts of the file can come from different sources. (If you've ever used BitTorrent, you may have noticed that when you download a file, BitTorrent can fetch it from multiple peers at once; this is the same idea.)
```

Comment:
Can we delete `you’d have to transfer the whole file in a single chunk every time someone requested it. If you had a huge file with a single CID, you’d have to transfer the whole file in a single chunk every time someone requested it. But instead, you can break it into blocks, and transfer it block by block. ` It's inaccurate that having blocks in the Merkle-DAG is necessary to do a "block" transfer of a file as you send the file in blocks (or segments) but not have them referenced in the dag. So let's just skip that explanation and move to the point that `Splitting it into blocks means that different parts of the file can come from different sources and BE authenticated quickly`.

Reply:
Done.